test/test_eos_vinet.cpp

```diff
@@ -0,0 +1,496 @@
+    Real rho0 = 8.93;
+    Real T0 = 298.0;
+    Real B0 = 1.3448466 * Mbcc_per_g;
+    Real BP0 = 4.956;
+    Real A0 = 5.19245e-05;
+    Real Cv0 = 0.383e-05 * Mbcc_per_g;
+    constexpr Real E0 = 0.0;
+    constexpr Real S0 = 5.05e-04 * Mbcc_per_g;
+    constexpr Real d2to40[39] = {0., 0.,      0.0000000000001,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 0.,      0.,
+                                 0., 1.0e-26, 0.,
+                                 0., 0.,      0.};
+    // Create the EOS
+    EOS host_eos = Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40);
+    EOS eos = host_eos.GetOnDevice();
+    eos.PrintParams();
+    eos.Finalize();
+
+    WHEN("Faulty/not set parameter rho0 is given") {
+      rho0 = -1.0;
+      REQUIRE_MAYBE_THROWS(Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40));
+      THEN("An error message should be written out") {
+        printf("Did catch negative rho0\n\n");
+      }
+    }
+    WHEN("Faulty/not set parameter T0 is given") {
+      T0 = -1.0;
+      REQUIRE_MAYBE_THROWS(Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40));
+      THEN("An error message should be written out") {
+        printf("Did catch negative T0\n\n");
+      }
+    }
+    WHEN("Faulty/not set parameter B0 is given") {
+      B0 = -1.0;
+      REQUIRE_MAYBE_THROWS(Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40));
+      THEN("An error message should be written out") {
+        printf("Did catch negative B0\n\n");
+      }
+    }
+    WHEN("Faulty/not set parameter BP0 is given") {
+      BP0 = 0.0;
+      REQUIRE_MAYBE_THROWS(Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40));
+      THEN("An error message should be written out") { printf("Did catch zero BP0\n\n"); }
+    }
+    WHEN("Faulty/not set parameter Cv0 is given") {
+      Cv0 = -1.0;
+      REQUIRE_MAYBE_THROWS(Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40));
+      THEN("An error message should be written out") {
+        printf("Did catch negative Cv0\n\n");
+      }
+    }
+    WHEN("Faulty/not set parameter A0 is given") {
+      A0 = -10000000.0;
+      REQUIRE_MAYBE_THROWS(Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40));
+      THEN("An error message should be written out") {
+        printf("Did catch negative A0\n\n");
+      }
+    }
+    //    WHEN("Reasonable parameters are given") {
+    //      REQUIRE_MAYBE_THROWS(Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40));
+    //      THEN("An error message should be written out") {
+    //        printf("This test should not pass\n\n");
+    //      }
+    //    }
```

Comment:
rather than commenting this out we should probably just delete this for now if we know it's all working.